src/models/CompanyNotification/Mapper.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  CompanyNewJobApplicationNotification,
+  ICompanyNewJobApplicationNotificationAttributes,
+  TCompanyNotification,
+  CompanyNotificationType
+} from "$models/CompanyNotification";
+import { CompanyNotification } from "$models";
+
+const typeMapper = {
```

Comment:
l final no me dejo typescript guardar la clase en el hash e instanciarla, se queja

Reply:
de que se queja?

igual imaginemos que no podemos usar un hash y su reverse, en ese caso usaria para ambos metodos switch statements con is instance of en el caso de toPersistenceModel